micro-rdk/src/common/data_manager.rs

```diff
@@ -27,6 +27,23 @@ use thiserror::Error;
 // the smaller amount of available RAM, we've halved it
 static MAX_SENSOR_CONTENTS_SIZE: usize = 32000;
 
+/// Allow for a C project using micro-RDK as a library to implement a callback to be run
+/// whenever data has successfully been uploaded. The callback should be identical in signature
+/// to the weak symbol declared here
+///
+/// Ex. for ESP-IDF project
+/// void data_upload_callback(void) {
```

Comment:
Name in this comment needs to be updated to `micro_rdk_data_manager_post_upload_hook`